asterius/src/Asterius/Types/SymbolMap.hs

```diff
@@ -75,6 +76,9 @@ newtype SymbolMap a = SymbolMap (IM.IntMap (EntitySymbol, a))
   deriving newtype (Eq, Semigroup, Monoid)
   deriving stock (Data)
 
+instance NFData a => NFData (SymbolMap a) where
+  rnf = rwhnf
```

Comment:
This is wrong. Forcing the `IntMap` to WHNF doesn't force its values. I think the TH instance generator should also works here even if `IntMap` data constructors are not exported. But in case TH doesn't work, the correct instance is folding over the `IntMap` and applying `rnf` to the values along the way.

Reply:
Right, I was sloppy here. Since `IntMap` is already an instance of `NFData` can't we also do
```haskell
instance NFData a => NFData (SymbolMap a) where
  rnf (SymbolMap m) = rnf m
```
?
If the `NFData` instance for `IntMap` is correct, I'd assume that it already does the traversal you describe, right?

Reply:
I just checked, `IntMap` already has `NFData` instance. We should be reusing that.

Reply:
Cool, I'll replace it with the two lines above then :slightly_smiling_face: (same for `SymbolSet`).